### ISSUE TITLE
All operations ignore worldedit blacklist

### ISSUE THREAD

**Issue by manuelgu** [SBPrime/AsyncWorldEdit-Premium#97]
Using version 3.4.1 of AWE and 6.0.3 of Worldedit, all worldedit operations ignore the blacklist from the worldedit configuration `limits.disallowed-blocks`

These are the permissions users have

```
worldedit.clipboard.copy (own)
AWE.user.cancel (own)
worldedit.navigation.* (own)
worldedit.generation.* (own)
worldedit.region.* (own)
worldedit.brush.* (own)
worldedit.biome.set (own)
worldedit.biome.list (own)
worldedit.biome.info (own)
worldedit.clipboard.clear (own)
worldedit.clipboard.paste (own)
worldedit.clipboard.rotate (own)
worldedit.clipboard.flip (own)
worldedit.history.clear (own)
worldedit.history.redo (own)
worldedit.history.undo (own)
worldedit.wand (own)
worldedit.wand.toggle (own)
worldedit.selection.chunk (own)
worldedit.selection.pos (own)
worldedit.selection.hpos (own)
worldedit.selection.contract (own)
worldedit.selection.expand (own)
worldedit.selection.shift (own)
worldedit.analysis.count (own)
worldedit.tool.replacer (own)
worldedit.tool.flood-fill (own)
worldedit.tool.farwand (own)
worldedit.tool.lrbuild (own)
worldedit.tool.info (own)
worldedit.tool.tree (own)
worldedit.fill (own)
worldedit.analysis.distr (own)
worldedit.fast (own)
```

**Comment by SBPrime** [comment]
AWE overrides the item black list. By default AWE has the same black list as the default WE config.

You have 2 options:
1. Remove the blacklist entry in AWE (that will force AWE to use the WE one)
2. Add a custom blacklist in AWE config.

**Comment by manuelgu** [comment]
I don't have blacklist entries in my AWE configuration file.

What format would a AWE blacklist have? Not listed in the wiki

It has worked fine in past versions of AWE, I didnt change any configuration values.

**Comment by manuelgu** [comment]
Looks like there is supposed to be a list of denied blocks at `awe.permissionGroups.<group>.worldedit.disallowedBlocks`. It is not there though, doesnt that mean it should default to worldedits list?

**Comment by SBPrime** [comment]
Send me the version number of AWE that you are using + your actual config.

**Comment by manuelgu** [comment]
```
>ver AsyncWorldEdit
[09:54:28 INFO]: AsyncWorldEdit version 3.4.1
[09:54:28 INFO]: This plugin attaches into WorldEdit and makes it asynchronous.
[09:54:28 INFO]: Website: https://github.com/SBPrime/AsyncWorldEdit-Premium/wiki
[09:54:28 INFO]: Author: SBPrime
```

http://hastebin.com/ixuhewihay.sm

**Comment by manuelgu** [comment]
I have manually added a disallowedBlocks list in the AWE config for testing purposes and noticed that its still possible to bypass that using the /repl tool and //copy //paste

**Comment by SBPrime** [comment]
**blockSet: false**

Change this to true. By default the black list works only for patterns.

**Comment by manuelgu** [comment]
> blockSet: false
> 
> Change this to true. By default the black list works only for patterns.

That fixes the issue with the /repl tool. Is there a bypass permission for that?

The issue that AWE doesn't use the default WorldEdit blacklist still persistst.

**Comment by SBPrime** [comment]
> Is there a bypass permission for that?

Sorry not at this moment but I can add this in the next version.

> The issue that AWE doesn't use the default WorldEdit blacklist still persistst.

Send me your current AWE and WE config. I'll try to reproduce and fix the issue.

**Comment by manuelgu** [comment]
> Sorry not at this moment but I can add this in the next version.

Thanks :+1: 

> Send me your current AWE and WE config. I'll try to reproduce and fix the issue.

WorldEdit: http://hastebin.com/sefekefumu.vala
AsynWorldEdit: http://hastebin.com/ofutapayij.sm

This setup should use the blacklist from WorldEdit, but it does not.

**Comment by manuelgu** [comment]
Any update?

**Comment by SBPrime** [comment]
Ad 1. The bug is fixed in v3.4.2

Ad 2. You can add an addition permission group in AWE config and change the black list behavior there. If you still think that there should be a "special" permission to disable the black list for block placing and for patterns please start a new issue. At the moment I won't be adding it.
